packages/snaps-controllers/src/cronjob/CronjobController.ts

```diff
@@ -267,24 +332,139 @@ export class CronjobController extends BaseController<
   }
 
   /**
-   * Unregister all jobs related to the given snapId.
+   * Schedule a background event.
+   *
+   * @param backgroundEventWithoutId - Background event.
+   * @returns An id representing the background event.
+   */
+  scheduleBackgroundEvent(
+    backgroundEventWithoutId: Omit<BackgroundEvent, 'id' | 'scheduledAt'>,
+  ) {
+    // removing milliseond precision and converting to UTC.
+    const scheduledAt = DateTime.fromJSDate(new Date())
+      .toUTC()
+      .toFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
+    const event = {
+      ...backgroundEventWithoutId,
+      id: nanoid(),
+      scheduledAt,
+    };
+    this.#setUpBackgroundEvent(event);
+    this.update((state) => {
+      state.events[event.id] = castDraft(event);
+    });
+
+    return event.id;
+  }
+
+  /**
+   * Cancel a background event.
+   *
+   * @param origin - The origin making the cancel call.
+   * @param id - The id of the background event to cancel.
+   * @throws If the event does not exist.
+   */
+  cancelBackgroundEvent(origin: string, id: string) {
+    assert(
+      this.state.events[id],
+      `A background event with the id of "${id}" does not exist.`,
+    );
+
+    assert(
+      this.state.events[id].snapId === origin,
+      'Only the origin that scheduled this event can cancel it',
+    );
+
+    const timer = this.#timers.get(id);
+    timer?.cancel();
+    this.#timers.delete(id);
+    this.#snapIds.delete(id);
+    this.update((state) => {
+      delete state.events[id];
+    });
+  }
+
+  /**
+   * A helper function to handle setup of the background event.
+   *
+   * @param event - A background event.
+   */
+  #setUpBackgroundEvent(event: BackgroundEvent) {
+    const date = new Date(event.date);
+    const now = new Date();
+    const ms = date.getTime() - now.getTime();
+
+    if (ms < 0) {
+      throw new Error('Cannot schedule an event in the past.');
+    }
+
+    const timer = new Timer(ms);
+    timer.start(() => {
+      this.#messenger
+        .call('SnapController:handleRequest', {
+          snapId: event.snapId,
+          origin: '',
+          handler: HandlerType.OnCronjob,
+          request: event.request,
+        })
+        .catch((error) => {
+          logError(error);
```

Comment:
Can this be covered?

Reply:
I've been trying the console calls are not being picked up in the jest spy.

Reply:
IMO not worth spending a lot of time trying to cover it. We have a bunch of these types of uncovered lines already 😅